### PR TITLE
NYS 168: Restrict MCP block content edit access to blocks linked to their managed senators

### PR DESCRIPTION
**Pull request by nathanielwoodland**
Note that the related hotfix in https://github.com/nysenate/NYSenate.gov-Website-D9/pull/226 revokes all MCP block content access. This is the long term fix.

### CODE REVIEW

**Comment by nathanielwoodland** [comment]
@routinet look I used `UsersHelper` for this! Glad you kept putting that in my face, since I've found there are helper methods in there that work very well for this (and make a lot of sense for NYS's internal logic).

But also ... don't look, since I didn't wrap `->getStorage('node')` in a try/catch block again. Since our last chat, I've found a lot of code in core (and contrib) that calls `->getStorage('node')`--or the storage ID of other core Drupal entity types--without any extra exception handling. Given how frequently this method will be called in custom code, I'd love to simplify our usage of `->getStorage()` to only worry about exception handling for non-core entity types. Let me know what you think.  Of course, I'm susceptible to arm twisting on this one (especially when the person doing the arm twisting is the tech lead on the project ....). :)

File: web/modules/custom/nys_senators/src/Access/McpBlockContentAccessControlHandler.php [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  /**
+   * {@inheritdoc}
+   */
+  protected function checkAccess(EntityInterface $entity, $operation, AccountInterface $account) {
+    // Grant MCPs access to edit content blocks tied to their senator.
+    $current_user = UsersHelper::resolveUser();
+    if (UsersHelper::isMcp($current_user)) {
+      $managed_senator_tids = UsersHelper::getManagedSenators($current_user);
+      $is_block_linked_to_managed_senator = $this->entityTypeManager
+        ->getStorage('node')
```
Comment (routinet) [inline web/modules/custom/nys_senators/src/Access/McpBlockContentAccessControlHandler.php:66]: `getStorage()` can throw an exception.  `try..catch` pls

Comment (routinet) [reply web/modules/custom/nys_senators/src/Access/McpBlockContentAccessControlHandler.php:66]: including it without logging is fine.  I'm just concerned about whatever weird edge case is going to pop up and trigger an uncaught exception.  I dislike it also, but... whachagonnado?  (also dislike a do-nothing `catch`, but I feel it is the lesser of two evils)

Comment (nathanielwoodland) [reply web/modules/custom/nys_senators/src/Access/McpBlockContentAccessControlHandler.php]: Great, all set @routinet

**Review by routinet — APPROVED** [review]

**Review by aheaphy — APPROVED** [review]

**Pull request merged**